File: app/src/main/java/io/github/kobakei/materialfabspeeddialexample/MainActivity.java
package io.github.kobakei.materialfabspeeddialexample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import io.github.kobakei.materialfabspeeddial.FabSpeedDial;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final FabSpeedDial fab = (FabSpeedDial) findViewById(R.id.fab);
        /*
        fab.addMenu(new FabSpeedDialMenu.Builder(this)
                .setItemId(1)
                .setTitle("Alarm")
                .setTitleColor(R.color.fab_mini_text)
                .setTitleBackgroundDrawable(R.drawable.label)
                .setDrawable(R.drawable.ic_action_alarm)
                .setDrawableTint(R.color.fab_mini_text)
                .setFabBackgroundColor(R.color.fab_mini)
                .setRippleColor(R.color.colorAccent)
                .build());
        fab.addMenu(new FabSpeedDialMenu.Builder(this)
                .setItemId(2)
                .setTitle("Camera")
                .setTitleColor(R.color.fab_mini_text)
                .setTitleBackgroundDrawable(R.drawable.label)
                .setDrawable(R.drawable.ic_action_camera)
                .setDrawableTint(R.color.fab_mini_text)
                .setFabBackgroundColor(R.color.fab_mini)
                .setRippleColor(R.color.colorAccent)
                .build());
        fab.addMenu(new FabSpeedDialMenu.Builder(this)
                .setItemId(3)
                .setTitle("Cut")
                .setTitleColor(R.color.fab_mini_text)
                .setTitleBackgroundDrawable(R.drawable.label)
                .setDrawable(R.drawable.ic_action_cut)
                .setDrawableTint(R.color.fab_mini_text)
                .setFabBackgroundColor(R.color.fab_mini)
                .setRippleColor(R.color.colorAccent)
                .build());*/
        fab.addOnMenuClickListener(new FabSpeedDial.OnMenuClickListener() {
            @Override
            public void onMenuClick(View view, int itemId) {
                Toast.makeText(MainActivity.this, "Click: " + itemId, Toast.LENGTH_SHORT).show();
            }
        });

        Button button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (fab.isShown()) {
                    fab.hide();
                } else {
                    fab.show();
                }
            }
        });
    }
}

File: app/src/main/java/io/github/kobakei/materialfabspeeddialexample/MainActivity.kt
package io.github.kobakei.materialfabspeeddialexample

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button1 = findViewById<Button>(R.id.button1)
        button1.setOnClickListener {
            val intent = Intent(this@MainActivity, SimpleActivity::class.java)
            startActivity(intent)
        }

        val button2 = findViewById<Button>(R.id.button2)
        button2.setOnClickListener {
            val intent = Intent(this@MainActivity, Simple2Activity::class.java)
            startActivity(intent)
        }

        val button3 = findViewById<Button>(R.id.button3)
        button3.setOnClickListener {
            val intent = Intent(this@MainActivity, CoordinatorActivity::class.java)
            startActivity(intent)
        }

        val button4 = findViewById<Button>(R.id.button4)
        button4.setOnClickListener {
            val intent = Intent(this@MainActivity, VectorActivity::class.java)
            startActivity(intent)
        }
    }
}
